service/service-mock/service-mock-soap/src/main/java/com/castlemock/service/mock/soap/project/converter/SoapPortConverter.java

```diff
@@ -50,41 +60,63 @@
+                                                     final boolean generateResponse) {
         final Map<String, WSDLDocument> documents = this.getDocuments(files, SoapResourceType.WSDL);
         return getResults(documents, generateResponse);
     }
 
     public Set<SoapPortConverterResult> getSoapPorts(final String location,
                                                      final boolean generateResponse,
-                                                     final boolean loadExternal){
+                                                     final boolean loadExternal) {
         try {
             final List<File> files = this.fileManager.uploadFiles(location);
             final Map<String, WSDLDocument> documents = this.getDocuments(files, SoapResourceType.WSDL);
             final Map<String, WSDLDocument> allDocuments = new HashMap<>(documents);
 
-            if(loadExternal){
+            if (loadExternal) {
                 documents.values()
                         .forEach(document -> loadExternal(location, document, allDocuments));
             }
 
             return getResults(allDocuments, generateResponse);
-        } catch (Exception e){
+        } catch (Exception e) {
             throw new RuntimeException(e.getMessage(), e);
         }
     }
 
     private Set<SoapPortConverterResult> getResults(final Map<String, WSDLDocument> documents,
-                                                    final boolean generateResponse){
+                                                    final boolean generateResponse) {
         try {
             return documents.entrySet().stream()
                     .map(documentEntry -> {
-                        final String name = documentEntry.getKey();
+                        DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
+                        String name = documentEntry.getKey();
                         final WSDLDocument wsdlDocument = documentEntry.getValue();
                         final Document document = wsdlDocument.getDocument();
+                        final String content = DocumentUtility.toString(document);
+                        try {
+                            DocumentBuilder builder = builderFactory.newDocumentBuilder();
+                            Document xmlDocument = builder.parse(new InputSource(new StringReader(content)));
+                            try {
+                                XPath xPath = XPathFactory.newInstance().newXPath();
+                                String path = "/definitions/portType/@name";
+                                Node node = (Node) xPath.compile(path).evaluate(xmlDocument, XPathConstants.NODE);
+                                name = node
+                                        .getNodeValue()
+                                        .replace("_Port", "")
+                                ;
+                            } catch (XPathExpressionException e) {
```

Comment:
Same case here. We should perhaps handle these instead of just printing the stacktrace.